server/util/disk/disk.go

```diff
@@ -75,20 +75,32 @@ func FileExists(fullPath string) (bool, error) {
 	}
 }
 
-type readerCloseWrapper struct {
-	reader io.Reader
-	closer io.Closer
+type SectionReaderCloser struct {
+	sectionReader *io.SectionReader
+	closer        io.Closer
 }
```

Comment:
You can embed the sectionReader in your SectionReaderCloser and methods you don't define yourself will fall-through to it. Since you only really need to define Close, this should be much shorter. (It's not quite subclassing just to be clear: https://golang.org/doc/effective_go#embedding)

For ex:

type SectionReaderCloser struct {
  *io.SectionReader
  closer io.Closer
}
func (w *SectionReaderCloser) Close() error {
	return w.closer.Close()
}

Reply:
Fixed!